src/ImageSharp/ColorProfiles/CieLab.cs

```diff
@@ -0,0 +1,182 @@
+        this.L = l;
+        this.A = a;
+        this.B = b;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CieLab"/> struct.
+    /// </summary>
+    /// <param name="vector">The vector representing the l, a, b components.</param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public CieLab(Vector3 vector)
+        : this()
+    {
+        this.L = vector.X;
+        this.A = vector.Y;
+        this.B = vector.Z;
+    }
+
+    /// <summary>
+    /// Gets the lightness dimension.
+    /// <remarks>A value usually ranging between 0 (black), 100 (diffuse white) or higher (specular white).</remarks>
+    /// </summary>
+    public readonly float L { get; }
+
+    /// <summary>
+    /// Gets the a color component.
+    /// <remarks>A value usually ranging from -100 to 100. Negative is green, positive magenta.</remarks>
+    /// </summary>
+    public readonly float A { get; }
+
+    /// <summary>
+    /// Gets the b color component.
+    /// <remarks>A value usually ranging from -100 to 100. Negative is blue, positive is yellow</remarks>
+    /// </summary>
+    public readonly float B { get; }
+
+    /// <summary>
+    /// Compares two <see cref="CieLab"/> objects for equality.
+    /// </summary>
+    /// <param name="left">The <see cref="CieLab"/> on the left side of the operand.</param>
+    /// <param name="right">The <see cref="CieLab"/> on the right side of the operand.</param>
+    /// <returns>
+    /// True if the current left is equal to the <paramref name="right"/> parameter; otherwise, false.
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static bool operator ==(CieLab left, CieLab right) => left.Equals(right);
+
+    /// <summary>
+    /// Compares two <see cref="CieLab"/> objects for inequality
+    /// </summary>
+    /// <param name="left">The <see cref="CieLab"/> on the left side of the operand.</param>
+    /// <param name="right">The <see cref="CieLab"/> on the right side of the operand.</param>
+    /// <returns>
+    /// True if the current left is unequal to the <paramref name="right"/> parameter; otherwise, false.
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static bool operator !=(CieLab left, CieLab right) => !left.Equals(right);
+
+    /// <inheritdoc/>
+    public override int GetHashCode() => HashCode.Combine(this.L, this.A, this.B);
+
+    /// <inheritdoc/>
+    public override string ToString() => FormattableString.Invariant($"CieLab({this.L:#0.##}, {this.A:#0.##}, {this.B:#0.##})");
+
+    /// <inheritdoc/>
+    public override bool Equals(object? obj) => obj is CieLab other && this.Equals(other);
+
+    /// <inheritdoc/>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public bool Equals(CieLab other) =>
+        this.L.Equals(other.L)
+        && this.A.Equals(other.A)
+        && this.B.Equals(other.B);
+
```

Comment:
```suggestion
        new Vector3(this.L, this.A, this.B) == new Vector3(other.L, other.A, other.B);
```
I don't know if this is faster, but at least it's less [machine code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGAOgDkBXfGKASzFwG4BYAKDMq0ASgwB2GVsxoBhCPgAOrADYsAyiwBu7GFz4DqNEeMkwaASXFtRudrt56AzOVjYAJhFFKAnuVwYoDGAY5NKsMAAy2MB8AN585AmUTgBmShDYweHkMeQA5jAYnOQAvvGJxClpGeQAgtl5BUWlvIlJ5KnpwQBC9fmFJWUJg+QA2gCyBQAWEK5mCkoAFBMY07PzAPLyEh64NDW5ubC4NhowFkqsope5AJQAusMV5MAQEErkAKIAjgzYSrgLUIRKLkCArFg3cgAXgAfMNWitWLtwjRvr9/gswZMWDRwjd4YkAGSE8iI3Y1VE/P4ArE4mr4lqtIkksk0LqU9E08FQNk3HiMoYC0bLVZzeSLEUzMVKTbbax7A5HE5nTyXa73R5OF5vT5U/5UQFhSLAUHcyGwgkJUQwADu5AAajAgtAHAtWeE0KTJki9p7WV1zVDyNa7Y7nVBXbSeR7TdieTVPVHefzikA==).

Reply:
Yep. The machine code diff is much larger than expected and, looking at the instructions, SIMDified.